pkg/iris/mapping.go

```diff
@@ -56,30 +60,28 @@ func (m *mapper) MapDeployments(deployments *appsv1.DeploymentList) ([]Discovery
 	var deploymentDiscoveryItems []DiscoveryItem
 	for _, deployment := range deployments.Items {
 		deployment.ClusterName = m.ClusterName
-		deploymentItem := DiscoveryItem{
-			ID:      deployment.Namespace + ":" + deployment.Name + "-" + deployment.ClusterName,
-			Scope:   "workspace/" + m.WorkspaceId,
-			Type:    "leanix.vsm.item-discovered.runtimeObject",
-			Source:  "kubernetes/" + m.ClusterName,
-			Time:    deployment.CreationTimestamp.String(),
-			Subject: "deployment/" + deployment.Name,
-			Data:    deployment,
-		}
+		id := fmt.Sprintf("%s:%s-%s", deployment.Namespace, deployment.Name, deployment.ClusterName)
+		Type := "leanix.vsm.item-discovered.runtimeObject"
+		Subject := fmt.Sprintf("deployment/%s", deployment.Name)
+
+		// common attributes
+		scope := fmt.Sprintf("workspace/%s", m.WorkspaceId)
+		Source := fmt.Sprintf("kubernetes/%s#%s", m.ClusterName, m.runId)
+		Time := deployment.CreationTimestamp.String()
+
+		deploymentItem := GenerateDiscoveryItem(id, scope, Type, Source, Time, Subject, deployment)
```

Comment:
> Also one more critical question; what is the benefit of having a Factory function if it simply create the struct but does not encapsulate any more logic? Couldn't we then just just the DiscoveryItem{...} call directly ?

Yeah agree, It can be encapsulated as per the mapping object. Will take care accordingly.